### PR TITLE
Avoid separated registering of qute: handlers

### PR DESCRIPTION
**Pull request by NoctuaNivalis**
I was looking at implementing a new qute: handler for downloads, so I was studying the handlers file. Imho, registration decorators are easier to maintain than a global constant.

### CODE REVIEW

**Comment by The-Compiler** [comment]
As discussed in IRC:

Using decorators is a good idea, but I'd prefer if they added the handlers to a global dict like before instead of having a class-level dict with a classmethod - makes things a bit simpler, and I don't think a global hurts here.

I'll also leave a few comments on lines.

File: qutebrowser/browser/network/qutescheme.py [hunk 1]
```diff
@@ -49,6 +49,15 @@ class QuteSchemeHandler(schemehandler.SchemeHandler):
 
     """Scheme handler for qute: URLs."""
 
+    handlers = dict()
```
Comment (The-Compiler) [inline qutebrowser/browser/network/qutescheme.py]: Please use `{}` instead of `dict()`

File: qutebrowser/browser/network/qutescheme.py [hunk 2]
```diff
@@ -49,6 +49,15 @@ class QuteSchemeHandler(schemehandler.SchemeHandler):
...
+
+    @classmethod
+    def addHandler(cls, name):
```
Comment (The-Compiler) [inline qutebrowser/browser/network/qutescheme.py]: Please use `snake_case` naming - `camelCase` is used in qutebrowser for overrides of Qt methods only.

File: qutebrowser/browser/network/qutescheme.py [hunk 3]
```diff
@@ -49,6 +49,15 @@ class QuteSchemeHandler(schemehandler.SchemeHandler):
...
+        """Add a handler to the qute: sheme."""
```
Comment (The-Compiler) [inline qutebrowser/browser/network/qutescheme.py]: sheme -> scheme

File: qutebrowser/browser/network/qutescheme.py [hunk 4]
```diff
@@ -45,6 +45,14 @@
 pyeval_output = ":pyeval was never called"
 
 
+HANDLERS = {}
+
+def add_handler(name):
+    """Add a handler to the qute: scheme."""
+    def namedecorator(function):
```
Comment (The-Compiler) [inline qutebrowser/browser/network/qutescheme.py]: Please also add a `@functools.wraps(function)` decorator to `namedecorator`.

File: qutebrowser/browser/network/qutescheme.py [hunk 5]
```diff
@@ -108,6 +116,7 @@ def set(self, win_id, sectname, optname, value):
             message.error(win_id, e)
 
 
+@addHandler('pyeval')
```
Comment (The-Compiler) [inline qutebrowser/browser/network/qutescheme.py]: This should be `add_handler` everywhere :wink:

Comment (NoctuaNivalis) [inline qutebrowser/browser/network/qutescheme.py]: Oh damn my quick edits.

**Comment by NoctuaNivalis** [comment]
I don't think there's a need for a `@functools.wraps(function)` here. My decorator rendered the actual function broken by returning `None` in stead of a modified function. I've let it return the unmodified function now so that it's more clear we don't modify the function, only register it.

This has the downside (or upside?) the handler functions can still be used elsewhere.

**Comment by The-Compiler** [comment]
Indeed - I find decorators with arguments quite hard to understand :wink:

Looks good to me - I'll merge this later today!

**Comment by The-Compiler** [comment]
There were some issues left, but I fixed them when merging:

- pep8 style issues: 3525659
- vulture now thought those functions were unused, so I added support to `run_vulture.py` for them: ac10fbc
- `qute:version` was missing a decorator (found via the vulture check): 939e95b

**Pull request merged**